sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -394,13 +415,37 @@ Ext.define('SlateTasksTeacher.controller.Dashboard', {
                 SectionID: me.getCourseSelector().getSelection().getId()
             });
 
-        me.doSaveStudentTask(studentTask, function() {
-            taskAssigner.close();
-            me.getStudentTasksStore().load({
-                id: studentTask.getId(),
-                addRecords: true
-            });
-            Ext.toast(student.getFullName() + ' was assigned task: ' + task.get('Title'));
+        me.doSaveStudentTask(studentTask, function(rec, success, request) {
+            var response,
+                message = [],
+                validationErrors, key;
+
+            if (success) {
+                taskAssigner.close();
+                me.getStudentTasksStore().load({
+                    id: studentTask.getId(),
+                    addRecords: true
+                });
+                Ext.toast(student.getFullName() + ' was assigned task: ' + task.get('Title'));
+            } else {
+                response = JSON.parse(request.getResponse().responseText);
```

Comment:
@nbey use `Ext.decode` instead of `JSON.parse`, if the response had `application/json` as a content-type though the jarvus-apikit wrapper would have already decoded the response and populated `response.responseData`, you shouldn't need to parse json here